File: src/me/riseremi/mreader/FileUtils.java
package me.riseremi.mreader;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 *
 * @author riseremi <riseremi at icloud.com>
 */
class FileUtils {

    protected static String getFileContent(String fileName) throws IOException {
        long start = System.currentTimeMillis();

        System.out.println("\nReading file...");

        InputStream reader = FileUtils.class.getResourceAsStream(fileName);
        BufferedReader br = new BufferedReader(new InputStreamReader(reader));

        String fileContent = "", line;
        while ((line = br.readLine()) != null) {
            fileContent += line;
        }
        System.out.println("Lapsed: " + (System.currentTimeMillis() - start) + " ms");

        return fileContent;
    }
}

File: src/main/java/me/riseremi/mreader/FileUtils.kt
package me.riseremi.mreader

import me.riseremi.utils.loadStream

/**
 * @author riseremi <riseremi at icloud.com>
 */
internal object FileUtils {

    fun getFileContent(fileName: String): String {
        val start = System.currentTimeMillis()

        println("\nReading file...")

        val inputStream = loadStream(fileName)
        val inputString = inputStream.bufferedReader().use { it.readText() }

//        val reader = FileUtils::class.java.getResourceAsStream(fileName)
//        val br = BufferedReader(InputStreamReader(reader))
//
//        val fileContent = StringBuilder()
//        var line = br.readLine()
//        while (line != null) {
//            line = br.readLine()
//            fileContent.append(line)
//        }
        println("Time spent: " + (System.currentTimeMillis() - start) + " ms")

        return inputString
    }
}
